### PR TITLE
Creation of enumeration reference for discrete math

### PR DESCRIPTION
**Pull request by arisdelacruz**
### Description

This is an attempt to create an enumeration reference for discrete math with further enhancement in mind, if needed.

### Type of Change

- Adding a new entry

### Checklist

- [ / ] All writings are my own.
- [ / ] My entry follows the Codecademy Docs style guide.
- [ / ] My changes generate no new warnings.
- [ / ] I have performed a self-review of my own writing and code.
- [ / ] I have checked my entry and corrected any misspellings.
- [ / ] I have made corresponding changes to the documentation if needed.
- [ / ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ / ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ / ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:51]:
```suggestion
The rule states that if there are `n(A)` ways to do `A` and `n(B)` ways to do `B`, then the number of ways to do A and B is `n(A) x n(B)`. This is true if the number of ways of doing `A` and `B` are independent; the number of choices for doing `B` is the same regardless of which choice is made for `A`.
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:27]:
```suggestion
Suppose there are 3 shirts (call them `A`, `B`, and `C`), and 4 pairs of pants (call them `w`, `x`, `y`, and `z`). There will be `3x4=12` possible outfits:
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:29]:
```suggestion
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:31]:
```suggestion
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:32]: Can we incase lines 33-35 in a [```plaintext code block](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#:~:text=Plaintext%3A%20%60%60%60plaintext)

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:45]: Incase line 45 into a ```tex code block.

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:55]:
```suggestion
There are `n(A) x n(B) x n(C)` ways to do `A` and `B` and `C`.
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:59]:
```suggestion
The rule states that if there are `n(A)` ways to do `A` and, distinct from them, `n(B)` ways to do `B`, then the number of ways to do `A` or `B` is `n(A) + n(B)`.
```

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (KTom101) [inline content/discrete-math/concepts/enumeration/enumeration.md:63]:
```suggestion
There are `n(A) + n(B) + n(C)` ways to do `A` or `B` or `C`.
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@arisdelacruz I've added some changes to this for the first review.

**Comment by arisdelacruz** [comment]
Thanks @KTom101

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (arisdelacruz) [reply content/discrete-math/concepts/enumeration/enumeration.md:32]: ayt

Noted with thanks

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (arisdelacruz) [reply content/discrete-math/concepts/enumeration/enumeration.md:45]: duly noted with thanks

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (arisdelacruz) [reply content/discrete-math/concepts/enumeration/enumeration.md]: did a commit; hope I got it right, hehe

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (arisdelacruz) [reply content/discrete-math/concepts/enumeration/enumeration.md]: same here, hope I got it right

**Review by KTom101 — APPROVED** [review]
@arisdelacruz I corrected a couple of things directly. Please wait for a second review.

**Comment by arisdelacruz** [comment]
@KTom101 

Noted with thanks

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (yangc95) [inline content/discrete-math/concepts/enumeration/enumeration.md:41]:
```suggestion
The formula (ꓴ is "AND", ꓵ is "OR"):
```

Comment (yangc95) [reply content/discrete-math/concepts/enumeration/enumeration.md:41]: mentioning this once would be good

File: content/discrete-math/concepts/enumeration/enumeration.md
Comment (yangc95) [inline content/discrete-math/concepts/enumeration/enumeration.md:19]:
```suggestion
**Enumeration** is another word for count. In the context of mathematics, enumerating is a type of counting given a specified event or situation. An enumeration is a complete, ordered listing of all the items in a collection. When speaking in the context of mathematics and computer science, listing all the elements of a set comes into play.

**Counting Theory** is a part of discrete mathematics that deals with counting the number of possible outcomes of a given event. Being an essential element of probability, it tries to answer an abstract question, "How many?"
```

I think your definition for counting theory is spot on! Enumerating is synonymous with counting and so I think in the context of discrete math (the topic), that should be mentioned. Then, it would be good to go into counting theory.

**Review by yangc95 — COMMENTED** [review]
@arisdelacruz done with the review! I like how you broke down each principle/rule in a painless way. I noticed there's also a subtraction and division rule. If you're up for it, would you add that in? Please let me know otherwise!

**Comment by arisdelacruz** [comment]
@yangc95

No prob. I'll look into them. I'll also add the foundational "Choices" on how one can preliminarily determine on how to combine numbers. These can make the entry more comprehensive

Thanks

**Comment by arisdelacruz** [comment]
@yangc95 

just a heads up

am finalizing the additional inputs

so interesting of a topic, hehe

good day

**Review by yangc95 — APPROVED** [review]

**Pull request merged**
